servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -1465,6 +1467,33 @@ private TypeSpec newServiceInterfaceSpec(final State state, final boolean blocki
                 .addStatement("return $L", blocking ? BLOCKING_METHOD_DESCRIPTORS : ASYNC_METHOD_DESCRIPTORS)
                 .build());
 
+        // generate default service methods
+        if (defaultServiceMethods) {
+            for (int i = 0; i < state.serviceProto.getMethodList().size(); i++) {
+                final MethodDescriptorProto methodProto = state.serviceProto.getMethodList().get(i);
+                final ClassName inClass = messageTypesMap.get(methodProto.getInputType());
+                final ClassName outClass = messageTypesMap.get(methodProto.getOutputType());
+                final String methodName = sanitizeIdentifier(methodProto.getName(), true);
+                final int methodIndex = i;
+                interfaceSpecBuilder.addMethod(newRpcMethodSpec(inClass, outClass, methodName,
+                        methodProto.getClientStreaming(),
+                        methodProto.getServerStreaming(),
+                        !blocking ? EnumSet.of(INTERFACE) : EnumSet.of(INTERFACE, BLOCKING),
+                        printJavaDocs, (__, c) -> {
+                            c.addModifiers(DEFAULT)
+                                    .addParameter(GrpcServiceContext, ctx)
+                                    .addStatement("throw new UnsupportedOperationException(\"Method " + methodName +
```

Comment:
1. gRPC services should use `GrpcStatusException` with `UNIMPLEMENTED` status, similar to official grpc-java to be compatible. Consider adding a test in `ProtocolCompatibilityTest` to make sure our default behavior is the same as grpc-java default behavior. In grpc-netty tests we can always use this new option, no need to create a new test module.
2. Async RPC that return `Single` or `Publisher` can not throw, they should return a failed async source, like `Single.failed(...)` or `Publisher.failed(...)`.
3. grpc-java uses `fullMethodName()` from method descriptor for exception message.

Reply:
Sounds good. I hammered this together to gather feedback on the option. I'll clean things up.

Reply:
FWIW, `fullMethodName` is an internal concoction by the grpc library. I was able to track down the source and am able to replicate.